src/main/java/io/fabric8/maven/docker/StopMojo.java

```diff
@@ -71,6 +72,41 @@
     @Parameter(property = "docker.stopNamePattern")
     private String stopNamePattern;
 
+    /**
+     * If true, the containers are not stopped right away, but when the build is finished (success or failed).
+     */
+    @Parameter(property = "docker.executeStopOnVMShutdown", defaultValue = "false")
+    private boolean executeStopOnVMShutdown;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
```

Comment:
All goals in this plugin override `executeInternal` method which contains goal related functionality. I'm not sure why you're overriding `execute` method. Is this possible to move this stop on VM shutdown logic to `executeInternal` method instead?